### ISSUE TITLE
introduce an Appdex monitor

### ISSUE THREAD

**Issue by kbilsted** [mvno/Okanshi#65]
Apdex provides three thresholds estimating end user satisfaction, satisfied, tolerating and frustrating.

    Satisfied: Response time less than or equal to T seconds.
    Tolerating: Response time between T seconds and 4T seconds.
    Frustrating: Response time greater than 4 T seconds.

The Apdex score between 0 and 1 is calculated using the following:

```( Satisfied + (Tolerating/2) ) / Total number of requests```

**Comment by kbilsted** [comment]
We never agreed on this, but I've implemented one now. It is awaiting PR #80

**Comment by kichristensen** [comment]
Sounds good. Should it be part of Okanshi or a separate package? What do you think?

**Comment by kbilsted** [comment]
its only 40 lines of code so I'd prefer it to be a part of the standard package. 

There are 2 issues worth discussing

* It only calculates based on the timer in use - ie controlled by the poller i assume? Other implementations use a longer spanning window. I'm not sure its a problem. Just raising it for debate
* I Imagine you can get an instance using `OkanshiMonitor.Apdex("foo", TimeSpan.FromSeconds(1))` this places it in the registry - but what if you then later on write `OkanshiMonitor.Apdex("foo", TimeSpan.FromSeconds(3))` ?

**Comment by kbilsted** [comment]
I cannot time it myself. It is the threshold that defines the acceptable response time

**Comment by kichristensen** [comment]
Yeah of course it is. Then we need to decide if it is same monitor or a different one. If we decide that it is different monitors then we could use a tag to differentiate them, and this information will be automatically propagated to the observer, that can decide what to do

**Comment by kbilsted** [comment]
That is not at all a bad idea to have the threshold be a tag..

**Comment by kbilsted** [comment]
Code is found at https://github.com/kbilsted/Okanshi/tree/pr_apdex_timer   I'd like your perspective before a PR

**Comment by kichristensen** [comment]
I added some comments your branch. Overall it looks great,
